src/execution_plan/ops/shared/create_functions.c

```diff
@@ -244,6 +245,21 @@ PendingProperties *ConvertPropertyMap(Record r, PropertyMap *map, bool fail_on_n
 				ErrorCtx_RaiseRuntimeException("Cannot merge node using null property value");
 			}
 		}
+
+		// emit an error and exit if we're trying to add
```

Comment:
I've notice `converted` is leaking in-case of an error, consider switching to a two loops logic:
```
1. validate expressions (can through an error)
2. allocate
3. assign
```

Reply:
We can restructure the function like that, but I don't think it will fix any existing leaks. In the event of an error on property N, we call `PendingPropertiesFree(converted)` and free the previously computed property values 0..N.